app/services/api/to_user_sessions_hash2.rb

```diff
@@ -0,0 +1,73 @@
+class Api::ToUserSessionsHash2
+  def initialize(form:, user:)
+    @form = form
+    @user = user
+  end
+
+  def call
+    return Failure.new(form.errors) if form.invalid?
+
+    delete_sessions(data.select(&:deleted))
+
+    Success.new(
+      {
+        upload: new_in_mobile_app,
```

Comment:
Not sure I like the mobile app part in the name. The endpoint is open to any type of client